### ISSUE TITLE
Creation of custom objects fails even when they are registered/created with `new_class`

### ISSUE THREAD

**Issue by leelavg** [kr8s-org/kr8s#400]
### Which project are you reporting a bug for?

kr8s

### What happened?

``` python
>>> from kr8s.objects import new_class
>>> StorageClientNS = new_class(
...     kind="StorageClient",
...     version="ocs.openshift.io/v1alpha1",
...     namespaced=True,
... )
>>> sc = StorageClientNS("temp-client", "default")
>>> sc.create()
[...]
  File "/root/.virtualenvs/update-416/lib/python3.12/site-packages/kr8s/_api.py", line 158, in call_api
    raise ServerError(kr8s._exceptions.ServerError: the object provided is unrecognized (must be of type StorageClient): Object 'Kind' is missing in '{"metadata": {"name": "temp-client", "namespace": "default"}}' ({"metadata": {"name": "temp-cl ...)
>>> sc.raw['kind'] = sc.kind
>>> sc.raw['apiVersion'] = sc.version
>>> sc.create()
>>> sc.raw
{'apiVersion': 'ocs.openshift.io/v1alpha1', 'kind': 'StorageClient', 'metadata': {'creationTimestamp': '2024-06-11T05:19:11Z', 'generation': 1, 'name': 'temp-client', 'namespace': 'default', 'resourceVersion': '133029', 'uid': '33b37505-cd5d-4a92-baac-05e8c81dbcbb'}}
```
- However, pre-created APIObjects doesn't have this issue. Returning to python after sometime and not able to debug (my knowledge is from pre-asyncio 😅), I believe the type creation `type(name, base, attrs)` seems to be working (untested) in async context but when `sync` is derived out of it, above issue might be hit.

Another small unrelated one, let me know if you need a new issue, during creation of objects from files auto creation of classes always set `namespaced=True` is it possible to infer the namespace from the marshalled spec and set accordingly?

Thanks a lot for the project, I wanted something more that kubectl and less (in the sense of verbosity) than client-go and kr8s hits the sport, is helpful so far.

**Comment by jacobtomlinson** [comment]
Thanks for raising this. I'm not sure this has anything to do with `asyncio`.

It looks to me like Kubernetes is more strict around validating the spec of custom resources being passed to it than it is for core objects.

For example I can create a Pod where the raw spec doesn't include the `kind` or `apiVersion` and it creates just fine.

```python
from kr8s.objects import Pod

po = Pod({'metadata': {'name': 'foo'}, 'spec': {'containers': [{'name': 'foo', 'image': 'nginx'}], 'restartPolicy': 'Always'}})
po.create()  # Works!
```

However for custom resources those values need to be explicitly set.

```python
from kr8s.objects import new_class

Sidecar = new_class("Sidecar", "networking.istio.io/v1beta1")
sc = Sidecar({'metadata': {'name': 'default', 'namespace': 'prod-us1'}, 'spec': {'egress': [{'hosts': ['./*']}]}})
sc.create()
```

```pytb
---------------------------------------------------------------------------
HTTPStatusError                           Traceback (most recent call last)
File ~/Projects/kr8s-org/kr8s/kr8s/_api.py:145, in Api.call_api(self, method, version, base, namespace, url, raise_for_status, stream, **kwargs)
    144 if raise_for_status:
--> 145     response.raise_for_status()
    146 yield response

File ~/miniconda3/envs/kr8s/lib/python3.11/site-packages/httpx/_models.py:761, in Response.raise_for_status(self)
    760 message = message.format(self, error_type=error_type)
--> 761 raise HTTPStatusError(message, request=request, response=self)

HTTPStatusError: Client error '400 Bad Request' for url 'https://127.0.0.1:34993/apis/networking.istio.io/v1beta1/namespaces/prod-us1/sidecars'
For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400

The above exception was the direct cause of the following exception:

ServerError                               Traceback (most recent call last)
Cell In[13], line 1
----> 1 sc.create()

File ~/Projects/kr8s-org/kr8s/kr8s/_async_utils.py:110, in run_sync.<locals>.run_sync_inner(*args, **kwargs)
    108 if inspect.iscoroutinefunction(coro):
    109     portal = Portal()
--> 110     return portal.call(wrapped)
    111 raise TypeError(f"Expected coroutine function, got {coro.__class__.__name__}")

File ~/Projects/kr8s-org/kr8s/kr8s/_async_utils.py:90, in Portal.call(self, func, *args, **kwargs)
     88 while not self._portal:
     89     pass
---> 90 return self._portal.call(func, *args, **kwargs)

File ~/miniconda3/envs/kr8s/lib/python3.11/site-packages/anyio/from_thread.py:288, in BlockingPortal.call(self, func, *args)
    273 def call(
    274     self,
    275     func: Callable[[Unpack[PosArgsT]], Awaitable[T_Retval] | T_Retval],
    276     *args: Unpack[PosArgsT],
    277 ) -> T_Retval:
    278     """
    279     Call the given function in the event loop thread.
    280 
   (...)
    286 
    287     """
--> 288     return cast(T_Retval, self.start_task_soon(func, *args).result())

File ~/miniconda3/envs/kr8s/lib/python3.11/concurrent/futures/_base.py:456, in Future.result(self, timeout)
    454     raise CancelledError()
    455 elif self._state == FINISHED:
--> 456     return self.__get_result()
    457 else:
    458     raise TimeoutError()

File ~/miniconda3/envs/kr8s/lib/python3.11/concurrent/futures/_base.py:401, in Future.__get_result(self)
    399 if self._exception:
    400     try:
--> 401         raise self._exception
    402     finally:
    403         # Break a reference cycle with the exception in self._exception
    404         self = None

File ~/miniconda3/envs/kr8s/lib/python3.11/site-packages/anyio/from_thread.py:217, in BlockingPortal._call_func(self, func, args, kwargs, future)
    214         else:
    215             future.add_done_callback(callback)
--> 217         retval = await retval_or_awaitable
    218 else:
    219     retval = retval_or_awaitable

File ~/Projects/kr8s-org/kr8s/kr8s/_objects.py:274, in APIObject.create(self)
    272 async def create(self) -> None:
    273     """Create this object in Kubernetes."""
--> 274     async with self.api.call_api(
    275         "POST",
    276         version=self.version,
    277         url=self.endpoint,
    278         namespace=self.namespace,
    279         data=json.dumps(self.raw),
    280     ) as resp:
    281         self.raw = resp.json()

File ~/miniconda3/envs/kr8s/lib/python3.11/contextlib.py:204, in _AsyncGeneratorContextManager.__aenter__(self)
    202 del self.args, self.kwds, self.func
    203 try:
--> 204     return await anext(self.gen)
    205 except StopAsyncIteration:
    206     raise RuntimeError("generator didn't yield") from None

File ~/Projects/kr8s-org/kr8s/kr8s/_api.py:158, in Api.call_api(self, method, version, base, namespace, url, raise_for_status, stream, **kwargs)
    156 if e.response.status_code >= 400 and e.response.status_code < 500:
    157     error = e.response.json()
--> 158     raise ServerError(
    159         error["message"], status=error, response=e.response
    160     ) from e
    161 elif e.response.status_code >= 500:
    162     raise ServerError(
    163         str(e), status=e.response.status_code, response=e.response
    164     ) from e

ServerError: the object provided is unrecognized (must be of type Sidecar): Object 'Kind' is missing in '{"metadata": {"name": "default", "namespace": "prod-us1"}, "spec": {"egress": [{"hosts": ["./*"]}]}}' ({"metadata": {"name": "default ...)
```

I think we should be able to fix this by updating the `.raw` property to always include those values.

**Comment by jacobtomlinson** [comment]
Also could you open a separate issue for the `namespaced=True` issue. Thanks for the kind words about the library, glad it's useful for you.

**Comment by jacobtomlinson** [comment]
This should now be fixed in `v0.16.0`

**Comment by leelavg** [comment]
> I'm not sure this has anything to do with
- ack, pls excuse, partially a human tendency to point onto something new that I don't know as a reason 😅 

> This should now be fixed in v0.16.0
- superfast, pulled it, it works.

> Also could you open a separate issue
- done #404
